### ISSUE TITLE
Tracee Docker Image: Clang version is too old

### ISSUE THREAD

**Issue by AlonZivony** [aquasecurity/tracee#1156]
I tried running tracee from the image in DockerHub and it failed because **clang** version was too old - _"detected clang version: 11 is older than required minimum version: 12"_

The clang version in the image should be updated so clients could use it out of the box.

**Comment by rafaeldtinoco** [comment]
and the message seems wrong (as 11 is not older than 12). I had noticed that yesterday also (during some tests).

**Comment by danielpacak** [comment]
Same thing on my end. Can be reproduced with the default Vagrantfile in the repo:

```console
$ docker run --name tracee --rm --privileged -it   -v /lib/modules/:/lib/modules/:ro   -v /usr/src:/usr/src:ro   -v /tmp/tracee:/tmp/tracee   aquasec/tracee:0.6.4
Unable to find image 'aquasec/tracee:0.6.4' locally
0.6.4: Pulling from aquasec/tracee
Digest: sha256:8f3952bccf4d42beb6c46bd9aea30ee9d99eef8539d3d316e93ebbf6dc46c08e
Status: Downloaded newer image for aquasec/tracee:0.6.4
Loaded 15 signature(s): [TRC-1 TRC-13  TRC-2 TRC-14 TRC-3 TRC-11 TRC-9 TRC-4 TRC-5 TRC-12 TRC-8 TRC-6 TRC-10 TRC-7]
2021/11/25 10:57:15 detected clang version: 11 is older than required minimum version: 12
```

```console
$ uname -a
Linux ubuntu-focal 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by danielpacak** [comment]
It seems that aquasec/tracee:0.6.4 was based off of alpine 3.14 (alpine:latest at build time). With Alpine 3.14 you'll get Clang 11.
One possible solution might be to pin Alpine to 3.15 in Dockerfile which will install Clang 12. Anyway pinning versions makes released artefacts much more predictable so I'd pin both Alpine based image and clang package versions.

**Comment by itaysk** [comment]
we install llvm 11 but require 12 since CORE https://github.com/aquasecurity/tracee/pull/759 (which fixed the tracee-ebpf dockerfile but not the tracee dockerfile).
we've actually hit and reported this bug in https://github.com/aquasecurity/tracee/issues/773 but ignored it.

before moving to alpine base, we installed llvm directly from llvm ppa with a specific version, which we should probably start doing again

**Comment by yanivagman** [comment]
> It seems that aquasec/tracee:0.6.4 was based off of alpine 3.14 (alpine:latest at build time). With Alpine 3.14 you'll get Clang 11.
> One possible solution might be to pin Alpine to 3.15 in Dockerfile which will install Clang 12. Anyway pinning versions makes released artefacts much more predictable so I'd pin both Alpine based image and clang package versions.

Moving to alpine 3.15 is a good and easy fix IMO

**Comment by danielpacak** [comment]
Resolved by #1173
